tests/contrib/celery/test_old_style_task.py

```diff
@@ -0,0 +1,35 @@
+import celery
+
+from .base import CeleryBaseTestCase
+from .utils import patch_task_with_pin
+
+
+class CeleryOldStyleTaskTest(CeleryBaseTestCase):
+    """Ensure Old Style Tasks are properly instrumented"""
+
+    def test_apply_async_previous_style_tasks(self):
+        # ensures apply_async is properly patched if Celery 1.0 style tasks
+        # are used even in newer versions. This should extend support to
+        # previous versions of Celery.
+        # Regression test: https://github.com/DataDog/dd-trace-py/pull/449
+        class CelerySuperClass(celery.task.Task):
+            abstract = True
+
+            @classmethod
+            def apply_async(cls, args=None, kwargs=None, **kwargs_):
+                return super(CelerySuperClass, cls).apply_async(args=args, kwargs=kwargs, **kwargs_)
+
+            def run(self, *args, **kwargs):
+                if 'stop' in kwargs:
+                    # avoid call loop
+                    return
+                CelerySubClass.apply_async(args=[], kwargs={"stop": True})
+
+        @patch_task_with_pin(pin=self.pin)
+        class CelerySubClass(CelerySuperClass):
+            pass
+
+        t = CelerySubClass()
+        t.run()
+        spans = self.tracer.writer.pop()
+        self.assertEqual(len(spans), 2)
```

Comment:
This is a change in the behavior, here why I'll put this refactoring PR in a WIP branch. Before we were expecting `4` and now we're tracing only half of the calls. The reason is because we don't instrument the Abstract class, so maybe `2` is the right output. I'll investigate more this part once this refactoring is merged.